File: examples/servers/dictionary/gateway_setup/teardown-gateway.ts
#!/usr/bin/env node

import {
  SSMClient,
  GetParameterCommand,
  DeleteParameterCommand,
} from "@aws-sdk/client-ssm";
import {
  BedrockAgentCoreControlClient,
  ListGatewayTargetsCommand,
  DeleteGatewayTargetCommand,
  DeleteGatewayCommand,
} from "@aws-sdk/client-bedrock-agentcore-control";

async function main() {
  const region = "us-west-2";

  // Get unique suffix for integration tests
  const suffix =
    "INTEG_TEST_ID" in process.env ? `-${process.env["INTEG_TEST_ID"]}` : "";

  // Get gateway info from SSM
  const ssmClient = new SSMClient({ region });

  let gatewayInfo: { gatewayId: string };
  try {
    const response = await ssmClient.send(
      new GetParameterCommand({
        Name: `LambdaMcpServer-Dictionary-Gateway${suffix}`,
      })
    );
    gatewayInfo = JSON.parse(response.Parameter!.Value!);
  } catch (error: any) {
    if (error.name === "ParameterNotFound") {
      console.log("Gateway parameter not found. Nothing to tear down.");
      return;
    }
    throw error;
  }

  const agentCoreClient = new BedrockAgentCoreControlClient({ region });

  // Delete all gateway targets
  const targets = await agentCoreClient.send(
    new ListGatewayTargetsCommand({
      gatewayIdentifier: gatewayInfo.gatewayId,
    })
  );

  for (const target of targets.items || []) {
    await agentCoreClient.send(
      new DeleteGatewayTargetCommand({
        gatewayIdentifier: gatewayInfo.gatewayId,
        targetId: target.targetId!,
      })
    );
    console.log(`Deleted target: ${target.targetId}`);
  }

  // Delete gateway
  await agentCoreClient.send(
    new DeleteGatewayCommand({
      gatewayIdentifier: gatewayInfo.gatewayId,
    })
  );

  // Delete SSM parameter
  await ssmClient.send(
    new DeleteParameterCommand({
      Name: `LambdaMcpServer-Dictionary-Gateway${suffix}`,
    })
  );

  console.log(`Gateway ${gatewayInfo.gatewayId} deleted successfully`);
}

main().catch(console.error);

File: examples/servers/zen/teardown_gateway.py
#!/usr/bin/env python3

import json
import os
import time
import boto3
from botocore.config import Config


def main():
    # Get unique suffix for integration tests
    suffix = f'-{os.environ["INTEG_TEST_ID"]}' if "INTEG_TEST_ID" in os.environ else ""

    # Get gateway info from SSM
    ssm_client = boto3.client("ssm", region_name="us-west-2")

    try:
        response = ssm_client.get_parameter(
            Name=f"LambdaMcpServer-Zen-Gateway{suffix}"
        )
        gateway_info = json.loads(response["Parameter"]["Value"])
        gateway_id = gateway_info["gatewayId"]
    except ssm_client.exceptions.ParameterNotFound:
        print("Gateway parameter not found. Nothing to tear down.")
        return

    # Delete all gateway targets
    retry_config = Config(
        retries={
            "max_attempts": 10,
            "mode": "standard",
        }
    )
    agentcore_client = boto3.client(
        "bedrock-agentcore-control", region_name="us-west-2", config=retry_config
    )

    targets = agentcore_client.list_gateway_targets(gatewayIdentifier=gateway_id)
    for target in targets["items"]:
        agentcore_client.delete_gateway_target(
            gatewayIdentifier=gateway_id, targetId=target["targetId"]
        )
        print(f"Deleted target: {target['targetId']}")

    # Wait for target deletions to propagate
    if targets["items"]:
        print("Waiting for target deletions to propagate...")
        time.sleep(5)

    # Delete gateway
    agentcore_client.delete_gateway(gatewayIdentifier=gateway_id)

    # Delete SSM parameter
    ssm_client.delete_parameter(Name=f"LambdaMcpServer-Zen-Gateway{suffix}")

    print(f"Gateway {gateway_id} deleted successfully")


if __name__ == "__main__":
    main()
